src/basic/process-util.c

```diff
@@ -351,9 +351,14 @@ int rename_process(const char name[]) {
 
                 mm = nn;
                 mm_size = nn_size;
-        } else
+        } else {
                 strncpy(mm, name, mm_size);
 
+                /* Update the end pointer, continuing regardless of any failure. */
+                if (prctl(PR_SET_MM, PR_SET_MM_ARG_END, (unsigned long) mm + l + 1, 0, 0) < 0)
+                        log_debug_errno(errno, "PR_SET_MM_ARG_END failed, proceeding without: %m");
```

Comment:
I'm not sure `strncpy` and `prctl` should be called when a process doesn't have enough privileges. I've just added a simple test that does something like this:
```sh
rename_process("abc")
fork()
....
setresuid(99, 99, 99)
rename_process("abcde")
```
. I see the following:
```sh
cmdline = <abc>
PR_SET_MM_ARG_END failed, proceeding without: Operation not permitted
cmdline = <abcdHOSTNAME=localhost.localdomain>
```
and I don't think that this is correct. The `if` branch checks that the result of `getuid` is equal to `0` before calling `strncpy` and `prctl`.